src/TodoContext.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import { getTodos, getUser } from './api';
+
+interface ContextValues {
+  todos: Todo[],
+  loading: boolean,
+  modalWindow: boolean,
+  selectedTodo: Todo | null,
+  modalLoading: boolean,
+  query: string,
+  filterOption: string,
+  setQuery: React.Dispatch<React.SetStateAction<string>>,
+  setFilterOption: React.Dispatch<React.SetStateAction<StateOption>>,
+  handleSelectedTodo: (todo: Todo) => void,
+  closeModal: () => void,
+  filterTodos: (
+    array: Todo[], filterQuery:string, selectedFilterOption: string
+  ) => Todo[],
+}
+
+export enum StateOption {
+  all = 'all',
+  active = 'active',
+  completed = 'completed',
+}
+
+export const TodoContext = React.createContext({} as ContextValues);
+
+export const TodoProvider: React.FC = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [modalLoading, setModalLoading] = useState(false);
+  const [modalWindow, setModalWindow] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterOption, setFilterOption]
+  = useState<StateOption>(StateOption.all);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(response => {
+        setTodos(response);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  const handleSelectedTodo = (todo: Todo) => {
+    setModalWindow(true);
+    setModalLoading(true);
+    getUser(todo.userId)
+      .then(response => setSelectedTodo({
+        ...todo,
+        user: response,
+      }))
+      .finally(() => setModalLoading(false));
+  };
+
+  const closeModal = () => {
+    setModalWindow(false);
+    setSelectedTodo(null);
+  };
+
+  const filterTodos = (
+    array: Todo[], filterQuery: string, selectedFilterOption: string,
+  ) => {
+    let filteredArray = [...array];
+    const normalizedQuery = filterQuery.toLowerCase();
+
+    if (normalizedQuery) {
+      filteredArray = filteredArray.filter(todo => todo.title
+        .toLowerCase().includes(normalizedQuery));
+    }
```

Comment:
Please check issues. You should trim search query
![image](https://github.com/mate-academy/react_dynamic-list-of-todos/assets/54936044/fd9a8004-fbfc-42e2-83cf-2b57df8775a9)

Reply:
I didn't do trim only because in working version if you typing spaces, it's not trimming